src/core_plugins/kibana/public/context/api/anchor.js

```diff
@@ -1,31 +1,41 @@
 import _ from 'lodash';
 
-import { addComputedFields } from './utils/fields';
-import { createAnchorQueryBody } from './utils/queries';
-
-
-async function fetchAnchor(es, indexPattern, uid, sort) {
-  const indices = await indexPattern.toIndexList();
-  const queryBody = addComputedFields(indexPattern, createAnchorQueryBody(uid, sort));
-  const response = await es.search({
-    index: indices,
-    body: queryBody,
-  });
-
-  if (_.get(response, ['hits', 'total'], 0) < 1) {
-    throw new Error('Failed to load anchor document.');
-  }
-
-  return Object.assign(
-    {},
-    response.hits.hits[0],
-    {
-      $$_isAnchor: true,
-    },
-  );
+import SearchSourceProvider from 'ui/courier/data_source/search_source';
+
+
+function fetchAnchorProvider(Private) {
+  const SearchSource = Private(SearchSourceProvider);
+
+  return async function fetchAnchor(indexPattern, uid, sort) {
+    const searchSource = new SearchSource()
+      .inherits(false)
```

Comment:
I'm trying to figure out what exactly this line accomplishes... Does the `SearchSource` inherit something else by default?